### PR TITLE
Floating-point and more jit tests

### PR DESCRIPTION
**Pull request by mihaibudiu**
Found one more bug in Calcite: https://issues.apache.org/jira/browse/CALCITE-5843!

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@Kixiron please review the 3-line JIT Rust patch.

File: docs/docs/sql/float.md [hunk 1]
```diff
@@ -2,15 +2,19 @@
 
 We support standard IEEE 754 floating point types.
 
-`double` is a 64-bit standard FP value.
+`double` is a 64-bit standard FP value.  Accepted synonyms are
+`float8` and `float64`.
```
Comment (Kixiron) [inline docs/docs/sql/float.md:6]: Floats aren’t ever specified by their byte length

**Review by Kixiron — APPROVED** [review]

File: docs/docs/sql/float.md (hunk 1 above)
Comment (mihaibudiu) [reply docs/docs/sql/float.md:6]: They are in SQL

Comment (mihaibudiu) [reply docs/docs/sql/float.md:6]: `float4` is so I can write Postgres tests... https://github.com/postgres/postgres/blob/master/src/test/regress/expected/float4.out

**Comment by mihaibudiu** [comment]
These tests pass on my machine, but fail in CI, and the logs don't say anything useful.

**Comment by ryzhyk** [comment]
Looks like a segfault?

**Comment by mihaibudiu** [comment]
where do you see that?

File: docs/docs/sql/float.md (hunk 1 above)
Comment (Kixiron) [reply docs/docs/sql/float.md:6]: Funky, if customers ever want `f16` or `f128` that could be problematic

**Comment by gz** [comment]
```           +test-sql *failed* | [WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /dbsp/sql-to-dbsp-compiler/SQL-compiler/target/surefire-reports/2023-07-13T01-33-01_711-jvmRun1.dumpstream
```

**Comment by gz** [comment]
```
           +test-sql *failed* | [ERROR] Tests run: 10, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 10.475 s <<< FAILURE! - in org.dbsp.sqlCompiler.compiler.postgres.PostgresFloat4Tests
           +test-sql *failed* | [ERROR] org.dbsp.sqlCompiler.compiler.postgres.PostgresFloat4Tests  Time elapsed: 10.112 s  <<< ERROR!
           +test-sql *failed* | java.lang.RuntimeException: Process failed with exit code 101
           +test-sql *failed* | 	at org.dbsp.util.Utilities.runProcess(Utilities.java:158)
           +test-sql *failed* | 	at org.dbsp.util.Utilities.compile(Utilities.java:172)
           +test-sql *failed* | 	at org.dbsp.util.Utilities.compileAndTestRust(Utilities.java:179)
           +test-sql *failed* | 	at org.dbsp.sqlCompiler.compiler.BaseSQLTests.runAllTests(BaseSQLTests.java:92)
           +test-sql *failed* | 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
           +test-sql *failed* | 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
           +test-sql *failed* | 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
           +test-sql *failed* | 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
           +test-sql *failed* | 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
           +test-sql *failed* | 	at org.junit.internal.runners.statements.RunAfters.invokeMethod(RunAfters.java:46)
           +test-sql *failed* | 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
           +test-sql *failed* | 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
           +test-sql *failed* | 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
           +test-sql *failed* | 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
           +test-sql *failed* | 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
           +test-sql *failed* | 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
           +test-sql *failed* | 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
           +test-sql *failed* | 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
           +test-sql *failed* | 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
           +test-sql *failed* | 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
           +test-sql *failed* | 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
```

exit code 101 is panic in rust

**Pull request merged**